### PR TITLE
Blog post: Typed Persistence

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: blog/_posts/2017-10-13-typed-persistence.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    PersistentActor.immutable[BlogCommand, BlogEvent, BlogState](
+      persistenceId = "abc",
+      initialState = BlogState.empty,
+      actions = PersistentActor.Actions { (ctx, cmd, state) ⇒ ??? },
+      applyEvent = (evt, state) ⇒ ???)
+
+}
+```
+
+There are 3 types:
```
Comment (raboof) [inline blog/_posts/2017-10-13-typed-persistence.md]: 'type parameters'?

File: blog/_posts/2017-10-13-typed-persistence.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+The function that process incoming commands is defined by the mandatory `commandHandler` parameter of the `Actions`.
+
+```scala
+  private val actions: Actions[BlogCommand, BlogEvent, BlogState] =
+    Actions { (ctx, cmd, state) ⇒
+      cmd match {
+        case AddPost(content, replyTo) ⇒
+          val evt = PostAdded(content.postId, content)
+          Persist[BlogEvent, BlogState](evt).andThen { state2 ⇒
```
Comment (raboof) [inline blog/_posts/2017-10-13-typed-persistence.md]: Are those type parameters really needed here or just added for clarity?

I think we planned to create constructors for these like `Effect.persist(...)` but I can't find a ticket for it - but perhaps it's good to wait until after the discussions in Austin on unifying some of the API's

File: blog/_posts/2017-10-13-typed-persistence.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+A command handler returns an `Effect` directive that defines what event or events, if any, to persist.
+
+* `Persist` will persist one single event
+* `PersistAll` will persist several events atomically, i.e. all events
+  are stored or none of them are stored if there is an error
+* `PersistNothing` no events are to be persisted, for example a read-only command
+* `Unhandled` the command is unhandled (not supported) in current state
+
+External side effects can be performed after successful persist with the `andThen` function. In the above example a reply is sent to the `replyTo`. Note that current state after applying the event is passed as parameter to the `andThen` function.
```
Comment (raboof) [inline blog/_posts/2017-10-13-typed-persistence.md]: perhaps 'new' or 'updated' instead of 'current'

File: blog/_posts/2017-10-13-typed-persistence.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    PersistentActor.persistentEntity[BlogCommand, BlogEvent, BlogState](
+      persistenceIdFromActorName = name => ShardingTypeName.name + "-" + name,
+      initialState = BlogState.empty,
+      actions = PersistentActor.Actions { (ctx, cmd, state) ⇒ ??? },
+      applyEvent = (evt, state) ⇒ ???)
+
+}
+```
+
+This also highlights a limitation with the current implementation. The `PersistentBehavior` can't be wrapped in other behaviors, such as `Actor.deferred`. The reason for that is that we are running the `PersistentBehavior` with an untyped `PersistentActor` and that doesn't allow such wrapping. We intend to re-implement this with a pure typed actor and then wrapping will be possible. The untyped journals, e.g. [akka-persistence-cassandra](https://github.com/akka/akka-persistence-cassandra), will still be possible to use as is. When this limitation is removed the above `PersistentActor.persistentEntity` will not be needed, because the `persistenceId` can be created from the actor name inside `deferred`.
```
Comment (raboof) [inline blog/_posts/2017-10-13-typed-persistence.md]: I can imagine this could be a little hard to follow if you're not more familiar with the solution yet... not sure how to improve though, good to mention it.

File: blog/_posts/2017-10-13-typed-persistence.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+## Feedback wanted
+
+The full source code of these examples, are available in [patriknw/akka-typed-blog](https://github.com/patriknw/akka-typed-blog).
+
+By this we are ending this series of blog posts about Akka Typed APIs for cluster and persistence. We expect a few more iterations to improve the APIs, so your feedback would be very valuable. We are currently working on other things but we will get back to Akka Typed soon and then we will focus on testing, documentation, general improvements of quality, and all [those small things](https://github.com/akka/akka/issues?q=is%3Aopen+is%3Aissue+label%3At%3Atyped). Help is of course welcome.
```
Comment (raboof) [inline blog/_posts/2017-10-13-typed-persistence.md]: Should we mention a Java DSL as one of the future topics explicitly?

Comment (johanandren) [reply blog/_posts/2017-10-13-typed-persistence.md]: That's the only thing I miss.

Comment (patriknw) [reply blog/_posts/2017-10-13-typed-persistence.md]: good point, I might turn it into a bullet list

File: blog/_posts/2017-10-13-typed-persistence.md (hunk 2 above)
Comment (patriknw) [reply blog/_posts/2017-10-13-typed-persistence.md]: they are not needed, and very ugly. removing

File: blog/_posts/2017-10-13-typed-persistence.md (hunk 4 above)
Comment (patriknw) [reply blog/_posts/2017-10-13-typed-persistence.md]: yes, I kind of agree but didn't want it to hide the fact. I re-read and I don't have any better phrasing. good thing is that this "advanced topic" is at the end so will probably not scare people from reading the post.

File: blog/_posts/2017-10-13-typed-persistence.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+
+The full source code of these examples, are available in [patriknw/akka-typed-blog](https://github.com/patriknw/akka-typed-blog).
+
+By this we are ending this series of blog posts about Akka Typed APIs for cluster and persistence. We expect a few more iterations to improve the APIs, so your feedback would be very valuable. We are currently working on other things but we will get back to Akka Typed soon and then we will focus on things like:
+
+* missing Java API, e.g. for Persistence
+* testing, testkit 
+* documentation
+* general improvements of quality
+* and all [those small things](https://github.com/akka/akka/issues?q=is%3Aopen+is%3Aissue+label%3At%3Atyped)
```
Comment (patriknw) [inline blog/_posts/2017-10-13-typed-persistence.md]: updated
